### PR TITLE
User patch

### PR DESCRIPTION
**Pull request by Jean-PhilippeD**
Add features to forward either event's user or session's user to Script context

### CODE REVIEW

File: api/core/script/script.exec.js [hunk 1]
```diff
@@ -5,17 +5,19 @@ module.exports = function(options) {
 
     // handle both direct call and scenario call
     var id = options.id || options.params.id;
+    if (options.scope && options.scope.user) var user = options.scope.user;
```
Comment (Pierre-Gilles) [inline api/core/script/script.exec.js:8]: You should create the user variable outside of the if scope, it's not really clear here where the user is defined

File: api/core/script/script.exec.js [hunk 2]
```diff
@@ -5,17 +5,19 @@ module.exports = function(options) {
...
+            return execCode(script.text, user);
         });
 };
 
 // execute the code
-function execCode(code) {
+function execCode(code, user) {
     return new Promise(function(resolve, reject) {
         try {
+            shared.sandbox.user = user;
```
Comment (Pierre-Gilles) [inline api/core/script/script.exec.js:20]: You cannot do that! You are modifying a global variable which is the same for all call to this function. So if two scripts are executed at the same time = problem !

**Comment by Jean-PhilippeD** [comment]
Ok I understand, do you have an idea to export user (or maybe other variable when calling the script through rest api ?) to the script ?

**Comment by Pierre-Gilles** [comment]
I think one good practice should be to pass to the sandbox a clone of all variables... I have an idea I will try to implement it ! 

I created an issue, it's referenced in the milestone for Gladys 3.4 which I'm trying to release this Sunday :)

**Comment by Jean-PhilippeD** [comment]
Hi,
In v 3.4 you've changed the way you declare the user regarding this PR.
It works fine when you pass from Controller, but when you come from a Scenario, you don't push user in the _options_ variable in the _core.script.exec.js_ so _context.user_ is not declared inside script.

In _script.exec.js_, don't you should add  ?
   ` if(!options.user && options.scope.user) options.user = options.scope.user;`
just after line 8.

In my PR, I was doing : 
`if (options.scope && options.scope.user) var user = options.scope.user;`
but it was not clear as you said.

**Comment by Pierre-Gilles** [comment]
Fixed here => https://github.com/GladysProject/Gladys/commit/e8428b11a743404041e269a756bfeae0fc7f2322 :)

**Pull request closed without merging**